l1-contracts/src/core/libraries/TimeMath.sol

```diff
@@ -11,170 +11,170 @@ type Slot is uint256;
 type Epoch is uint256;
 
 library SlotLib {
-  function toTimestamp(Slot a) internal pure returns (Timestamp) {
-    return Timestamp.wrap(Slot.unwrap(a) * Constants.AZTEC_SLOT_DURATION);
+  function toTimestamp(Slot _a) internal pure returns (Timestamp) {
+    return Timestamp.wrap(Slot.unwrap(_a) * Constants.AZTEC_SLOT_DURATION);
   }
 
-  function fromTimestamp(Timestamp a) internal pure returns (Slot) {
-    return Slot.wrap(Timestamp.unwrap(a) / Constants.AZTEC_SLOT_DURATION);
+  function fromTimestamp(Timestamp _a) internal pure returns (Slot) {
+    return Slot.wrap(Timestamp.unwrap(_a) / Constants.AZTEC_SLOT_DURATION);
   }
 
-  function positionInEpoch(Slot a) internal pure returns (uint256) {
-    return Slot.unwrap(a) % Constants.AZTEC_EPOCH_DURATION;
+  function positionInEpoch(Slot _a) internal pure returns (uint256) {
+    return Slot.unwrap(_a) % Constants.AZTEC_EPOCH_DURATION;
   }
 
-  function unwrap(Slot a) internal pure returns (uint256) {
-    return Slot.unwrap(a);
+  function unwrap(Slot _a) internal pure returns (uint256) {
+    return Slot.unwrap(_a);
   }
 }
 
 library EpochLib {
-  function toSlots(Epoch a) internal pure returns (Slot) {
-    return Slot.wrap(Epoch.unwrap(a) * Constants.AZTEC_EPOCH_DURATION);
+  function toSlots(Epoch _a) internal pure returns (Slot) {
+    return Slot.wrap(Epoch.unwrap(_a) * Constants.AZTEC_EPOCH_DURATION);
   }
 
-  function toTimestamp(Epoch a) internal pure returns (Timestamp) {
-    return SlotLib.toTimestamp(toSlots(a));
+  function toTimestamp(Epoch _a) internal pure returns (Timestamp) {
+    return SlotLib.toTimestamp(toSlots(_a));
   }
 
-  function fromTimestamp(Timestamp a) internal pure returns (Epoch) {
+  function fromTimestamp(Timestamp _a) internal pure returns (Epoch) {
     return Epoch.wrap(
-      Timestamp.unwrap(a) / (Constants.AZTEC_EPOCH_DURATION * Constants.AZTEC_SLOT_DURATION)
+      Timestamp.unwrap(_a) / (Constants.AZTEC_EPOCH_DURATION * Constants.AZTEC_SLOT_DURATION)
     );
   }
 
-  function unwrap(Epoch a) internal pure returns (uint256) {
-    return Epoch.unwrap(a);
+  function unwrap(Epoch _a) internal pure returns (uint256) {
+    return Epoch.unwrap(_a);
   }
 }
 
 using SlotLib for Slot;
 using EpochLib for Epoch;
 
-function addTimestamp(Timestamp a, Timestamp b) pure returns (Timestamp) {
-  return Timestamp.wrap(Timestamp.unwrap(a) + Timestamp.unwrap(b));
+function _addTimestamp(Timestamp _a, Timestamp _b) pure returns (Timestamp) {
```

Comment:
Not sure a rule where free functions need to be _ makes sense? They will be used in other files so I dont see them as "private"

Reply:
Ye, taking a look at altering the rule to see how simple it is, hopefully straight forward.

Reply:
https://github.com/LHerskind/solhint/commit/82c456c9a7c2c3a254d8de3a3c32722bc0fce699